### PR TITLE
Bug fix for no flags and combination of another PR #114

### PR DESCRIPTION
**Pull request by info-rchitect**
We don't use the native debugger flags for SMT7 and I recently disabled them all to save test time.  When I did this it broke origen_testers.

Combining this [PR](https://github.com/Origen-SDK/origen_testers/pull/114)

### CODE REVIEW

**Comment by info-rchitect** [comment]
@ginty I do not understand why the build failed?

~~~shell
cp list/j750/referenced.list approved/j750/referenced.list && cp list/j750_hpt/referenced.list approved/j750_hpt/referenced.list && cp list/ultraflex/referenced.list approved/ultraflex/referenced.list && cp list/v93k/referenced.list approved/v93k/referenced.list
[ERROR]      13.787[0.000]   || 
[ERROR]      13.787[0.000]   ||     FFFFFF     AA       II   LL
[ERROR]      13.787[0.000]   ||     FF        AAAA      II   LL
[ERROR]      13.787[0.000]   ||     FFFFF    AA  AA     II   LL
[ERROR]      13.787[0.000]   ||     FF      AAAAAAAA    II   LL
[ERROR]      13.787[0.000]   ||     FF     AA      AA   II   LL
[ERROR]      13.788[0.000]   ||     FF    AA        AA  II   LLLLLL
[ERROR]      13.788[0.000]   || 
~~~

I ran the command to copy the new files and then ran `origen save changed`.  Shouldn't this have worked?

**Comment by ginty** [comment]
Normally yes, but not for changes in regression.list within this test suite since it gets moved from its generated location to avoid the different test cases overwriting it.
I think the line in your snippet above `cp list/j750...` is the command you need to execute to take care of this.

**Comment by info-rchitect** [comment]
@ginty I did that command a couple times and don't see it making new files known to Git:

~~~shell
server03:origen_testers $ cp list/j750/referenced.list approved/j750/referenced.list && cp list/j750_hpt/referenced.list approved/j750_hpt/referenced.list && cp list/ultraflex/referenced.list approved/ultraflex/referenced.list && cp list/v93k/referenced.list approved/v93k/referenced.list
server03:origen_testers $ git status
On branch bug_fix_for_no_flags
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   Gemfile
        modified:   lib/origen_testers/test/interface.rb

no changes added to commit (use "git add" and/or "git commit -a")
~~~

**Comment by info-rchitect** [comment]
@ginty I think i see the issue.  There is no referenced list.

~~~shell
cp: cannot stat `list/referenced.list': No such file or directory
server03:origen_testers $ ls -alrt list/
total 40
-rw-r--r--  1 user design  299 Jan 30 06:47 j750.list
-rw-r--r--  1 user design   48 Jan 30 06:47 pxie6570.list
-rw-r--r--  1 user design  103 Jan 30 06:47 regression.list
drwxr-xr-x  2 user design   80 Mar  1 09:12 j750/
drwxr-xr-x  2 user design   80 Mar  1 09:12 j750_hpt/
drwxr-xr-x  2 user design   80 Mar  1 09:12 ultraflex/
drwxr-xr-x  2 user design   80 Mar  1 09:12 v93k/
drwxr-xr-x  2 user  design   80 Mar  1 09:12 v93k_multiport/
drwxr-xr-x  2 user design   80 Mar  1 09:12 v93k_enable_flow/
drwxr-xr-x  2 user  design   80 Mar  1 09:12 v93k_disable_flow/
drwxr-xr-x  2 user design   80 Mar  1 09:12 v93k_limits_file/
~~~

**Comment by ginty** [comment]
Not sure then, you'll need to investigate. In principle this line or something like it should work:

~~~
cp list/j750/referenced.list approved/j750/referenced.list && cp list/j750_hpt/referenced.list approved/j750_hpt/referenced.list && cp list/ultraflex/referenced.list approved/ultraflex/referenced.list && cp list/v93k/referenced.list approved/v93k/referenced.list
~~~

Maybe if only a few lists have been updated then it fails because a file doesn't exist and then doesn't continue with the rest.

Basically for these test cases only, it moves the list from its default location so the built in Origen save command doesn't work - https://github.com/Origen-SDK/origen_testers/blob/master/examples/program_generator.rb#L16

**Comment by info-rchitect** [comment]
@ginty I think the issue is a bug in the console output.

~~~shell
[INFO]       11.875[0.083]   || Log file written to: /home/travis/build/Origen-SDK/origen_testers/log/program_generator/dut/v93k_multiport/prb1.txt
[INFO]       11.992[0.117]   || Writing... prb1.tf
[INFO]       12.173[0.181]   || Writing... prb1.pmfl
[INFO]       12.178[0.005]   || Writing... prb1.aiv
[INFO]       12.190[0.011]   || Writing... prb1_vars.tf
[INFO]       12.202[0.013]   || Referenced pattern list written to: list/referenced.list
[INFO]       12.204[0.002]   || *** CHANGE DETECTED *** To update the reference:  cp list/referenced.list approved/v93k_multiport/referenced.list
[INFO]       12.205[0.000]   || tkdiff approved/v93k_multiport/referenced.list list/referenced.list &
~~~

Notice how it says the referenced pattern list is `list/references.list`?  Here is what i see in my list dir:

~~~shell
server03:origen_testers $ ls list
j750/  j750_hpt/  j750.list  pxie6570.list  regression.list  ultraflex/  v93k/  v93k_disable_flow/  v93k_enable_flow/  v93k_limits_file/  v93k_multiport/
server03:origen_testers $ ls list/v93k
referenced.list
~~~

There is a `regression.list` and and a `v93k/referenced.list`.  Should I be copying  `v93k/referenced.list` to `approved/v93k_multiport/referenced.list`?

**Comment by info-rchitect** [comment]
OK so I made a commit with some manually copied files and now have one error:

~~~shell
[INFO]       13.901[0.004]   || Generating... prb1.rb
[INFO]       13.985[0.084]   || Log file written to: /home/travis/build/Origen-SDK/origen_testers/log/program_generator/dut/v93k_global/prb1.txt
[INFO]       13.988[0.004]   || Generating... prb2.rb
[INFO]       14.002[0.013]   || Log file written to: /home/travis/build/Origen-SDK/origen_testers/log/program_generator/dut/v93k_global/prb2.txt
[INFO]       14.125[0.123]   || Writing... prb1.tf
[INFO]       14.300[0.175]   || *** CHANGE DETECTED *** To update the reference:  cp output/v93k_global/testflow/mfh.testflow.group/prb1.tf approved/v93k_global/prb1.tf
[INFO]       14.300[0.000]   || tkdiff approved/v93k_global/prb1.tf output/v93k_global/testflow/mfh.testflow.group/prb1.tf &
[INFO]       14.300[0.000]   || **********************************************************************
[INFO]       14.308[0.008]   || Writing... prb2.tf
[INFO]       14.343[0.035]   || Writing... global.pmfl
[INFO]       14.346[0.003]   || *** CHANGE DETECTED *** To update the reference:  cp output/v93k_global/vectors/global.pmfl approved/v93k_global/global.pmfl
[INFO]       14.346[0.000]   || tkdiff approved/v93k_global/global.pmfl output/v93k_global/vectors/global.pmfl &
[INFO]       14.347[0.000]   || **********************************************************************
[INFO]       14.348[0.002]   || Writing... global.aiv
[INFO]       14.352[0.003]   || *** CHANGE DETECTED *** To update the reference:  cp output/v93k_global/global.aiv approved/v93k_global/global.aiv
[INFO]       14.352[0.000]   || tkdiff approved/v93k_global/global.aiv output/v93k_global/global.aiv &
[INFO]       14.352[0.000]   || **********************************************************************
[INFO]       14.354[0.002]   || Writing... global_vars.tf
[INFO]       14.369[0.015]   || Referenced pattern list written to: list/referenced.list
[INFO]       14.372[0.003]   || *** CHANGE DETECTED *** To update the reference:  cp list/referenced.list approved/v93k_global/referenced.list
[INFO]       14.372[0.000]   || tkdiff approved/v93k_global/referenced.list list/referenced.list &
[INFO]       14.372[0.000]   || **********************************************************************
[INFO]       14.372[0.000]   || **********************************************************************
[INFO]       14.372[0.000]   || Total patterns:   96
[INFO]       14.372[0.000]   || Total vectors:    30955
[INFO]       14.372[0.000]   || Total duration:   0.893048
[INFO]       14.372[0.000]   || New patterns:     0
[WARNING]    14.372[0.000]   || Changed patterns: 1
[INFO]       14.373[0.000]   || 
[INFO]       14.373[0.000]   || Total files:      100
[INFO]       14.373[0.000]   || New files:        0
[INFO]       14.373[0.000]   || Changed files:    3
[INFO]       14.373[0.000]   || 
[INFO]       14.373[0.000]   || To accept all of these changes run:
[INFO]       14.373[0.000]   ||   origen save changed
[INFO]       14.373[0.000]   || **********************************************************************
To approve any diffs in the reference.list files run the following command:
cp list/j750/referenced.list approved/j750/referenced.list && cp list/j750_hpt/referenced.list approved/j750_hpt/referenced.list && cp list/ultraflex/referenced.list approved/ultraflex/referenced.list && cp list/v93k/referenced.list approved/v93k/referenced.list
[ERROR]      14.373[0.000]   || 
[ERROR]      14.373[0.000]   ||     FFFFFF     AA       II   LL
[ERROR]      14.373[0.000]   ||     FF        AAAA      II   LL
[ERROR]      14.373[0.000]   ||     FFFFF    AA  AA     II   LL
[ERROR]      14.373[0.000]   ||     FF      AAAAAAAA    II   LL
[ERROR]      14.374[0.000]   ||     FF     AA      AA   II   LL
[ERROR]      14.374[0.000]   ||     FF    AA        AA  II   LLLLLL
[ERROR]      14.374[0.000]   || 
~~~

I do not have an `output/v93k_global` directory to copy any files from:

~~~shell
server:origen_testers $ ls output/
j750/  j750_hpt/  OrigenTesters::LabVIEWBasedTester::Pxie6570/  ultraflex/  v93k/  v93k_disable_flow/  v93k_enable_flow/  v93k_limits_file/  v93k_multiport/
~~~

**Comment by priyavadan** [comment]
@info-rchitect please close this PR. A fix has been implemented in PR #131
